shared/pkg/scanner/trivy/local.go

```diff
@@ -0,0 +1,256 @@
+		case utils.SBOM:
+			trivySourceType = artifact.TargetSBOM
+		default:
+			a.logger.Infof("Skipping analyze unsupported source type: %s", sourceType)
+			a.resultChan <- &scanner.Results{
+				Matches: nil, // empty results,
+				ScannerInfo: scanner.Info{
+					Name: ScannerName,
+				},
+			}
+			return
+		}
+
+		err := artifact.Run(context.TODO(), trivyOptions, trivySourceType)
+		if err != nil {
+			a.setError(fmt.Errorf("failed to generate SBOM: %w", err))
+			return
+		}
+
+		a.logger.Infof("Sending successful results")
+		a.resultChan <- a.CreateResult(output.Bytes())
+	}()
+
+	return nil
+}
+
+func getCVSSesFromVul(vCvss trivyDBTypes.VendorCVSS) []scanner.CVSS {
+	// TODO(sambetts) Work out how to include all the
+	// CVSS's by vendor ID in our report, for now only
+	// collect one of each version.
+	cvsses := []scanner.CVSS{}
+	v2Collected := false
+	v3Collected := false
+	for _, cvss := range vCvss {
+		if cvss.V3Vector != "" && !v3Collected {
+			cvsses = append(cvsses, scanner.CVSS{
+				Version: "3.1",
+				Vector:  cvss.V3Vector,
+				Metrics: scanner.CvssMetrics{
+					BaseScore: cvss.V3Score,
+				},
+			})
+			v3Collected = true
+		}
+		if cvss.V2Vector != "" && !v2Collected {
+			cvsses = append(cvsses, scanner.CVSS{
+				Version: "2.0",
+				Vector:  cvss.V2Vector,
+				Metrics: scanner.CvssMetrics{
+					BaseScore: cvss.V2Score,
+				},
+			})
+			v2Collected = true
+		}
+	}
+	return cvsses
+}
+
+func (a *LocalScanner) CreateResult(trivyJSON []byte) *scanner.Results {
+	result := &scanner.Results{
+		Matches: nil, // empty results,
+		ScannerInfo: scanner.Info{
+			Name: ScannerName,
+		},
+	}
+
+	var report trivyTypes.Report
+	err := json.Unmarshal(trivyJSON, &report)
+	if err != nil {
+		a.logger.Error(err)
+		result.Error = err
+		return result
+	}
+
+	matches := []scanner.Match{}
+	for _, result := range report.Results {
+		for _, vul := range result.Vulnerabilities {
+			typ, err := getTypeFromPurl(vul.Ref)
+			if err != nil {
+				a.logger.Error(err)
+				typ = ""
+			}
+
+			cvsses := getCVSSesFromVul(vul.CVSS)
+
+			fix := scanner.Fix{}
+			if vul.FixedVersion != "" {
+				fix.Versions = []string{
+					vul.FixedVersion,
+				}
+			}
+
+			kbVul := scanner.Vulnerability{
+				ID:          vul.VulnerabilityID,
+				Description: vul.Description,
+				Links: []string{
+					vul.PrimaryURL,
+				},
+				Distro: scanner.Distro{
+					Name:    report.Metadata.OS.Family,
+					Version: report.Metadata.OS.Name,
+				},
+				CVSS:     cvsses,
+				Fix:      fix,
+				Severity: vul.Severity,
+				Package: scanner.Package{
+					Name:    vul.PkgName,
+					Version: vul.InstalledVersion,
+					PURL:    vul.Ref,
+					Type:    typ,
+				},
+				LayerID: vul.Layer.Digest,
+				Path:    vul.PkgPath,
+			}
+
+			matches = append(matches, scanner.Match{
+				Vulnerability: kbVul,
+			})
+		}
+	}
+
+	a.logger.Infof("Found %d vulnerabilities", len(matches))
+
+	source := scanner.Source{
+		Name: report.ArtifactName,
```

Comment:
what about the `hash`?

Reply:
updated to handle this like the grype scanner does